core/util/activesearch.py

```diff
@@ -0,0 +1,145 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+import urllib.parse
+import html
+
+class main:
+
+		def __init__(self, q, limit=15):
+			""" activesearchresults.com search engine
+
+					q         : query for search
+					limit     : maximum result count
+			"""
+			self.framework = main.framework
+			self.q = q
+			self._max = limit
+			self._cookie = ''
+			self._rawhtml = ''
+			self._clubbedrows = []
+			self._rows = []
+			self._links = []
+			self._links_with_data = []
+			self._pageno = 0
+			self.active_search = 'www.activesearchresults.com'
+			
+
+		def run_crawl(self):
+			payload = { 
+				'wordsall': self.q,
+				'asrtech' : 'yes',
+				'perpage' : '100',
+				'adultfilter': '5',
+				'buttonall': ''
+				}
+			header = {
+				'Cookie' : f'{self._cookie}'
+				}
+
+			self.framework.verbose('Searching the activesearchresults domain...')
+
+			while True:
+				self._pageno += 1
+				url = f'https://{self.active_search}/searchsubmit.php?pageno={self._pageno}'
+
+				try:
+					req = self.framework.request(url=url,
+						method='POST',
+						data=payload,
+						headers=header
+						)
+				except:
+					self.framework.error('[ACTIVESEARCH] ConnectionError')
+					self.framework.error('activesearchresults is missed!')
+					return
+
+				self._rawhtml += req.text
+
+				if self._pageno==1:
+					max_pages = list(map(int, re.findall(
+						r"<a href='/searchsubmit.php\?pageno=(\d*?)'>LAST",
+						req.text)))
+
+					if len(max_pages)==0:
+						break
+
+					else:
+						max_pages = max_pages[0]
+
+					self._cookie = req.headers.get('Set-Cookie')
+
+
+				if self._pageno > max_pages or self._pageno*100>self._max:
+					break
+					    
+			self._clubbedrows = '\n'.join(re.findall(r'<small>(.*?)</small>',
+				self._rawhtml,
+				flags=re.DOTALL))
+
+			self._rows = re.findall(r'<a.*?<hr>', 
+				self._clubbedrows, 
+				flags=re.DOTALL)
+
+			self._links = re.findall(r"<a href='(.*?)'.*?>.*?<br>",
```

Comment:
don't use *. apply it for all.

Reply:
the titles have no fixed format though right, I have to use .+? at best

Reply:
Let me try using reglib

Reply:
No

Reply:
why not. For the titles Im just getting the inner text within the a tag using .+? Is there a better way to get inner text?

Reply:
Why not?
![image](https://user-images.githubusercontent.com/31902891/113428264-5a521c80-938b-11eb-9fe9-2e8ea7a21729.png)
Got it?:
![image](https://user-images.githubusercontent.com/31902891/113428562-de0c0900-938b-11eb-8f8d-92c88fedc225.png)
Another example:
![image](https://user-images.githubusercontent.com/31902891/113428701-1c092d00-938c-11eb-98fa-f4371dc9c943.png)

Reply:
oh no no I am talking about the titles

Reply:
"the titles have no fixed format though right" is what I said

Reply:
Use bs4 to extract the data.

Reply:
cool